airflow/tests/test_unit.py

```diff
@@ -95,8 +95,17 @@ def test_dag_total_tasks(aggregator, task_instance):
     aggregator.assert_metric('airflow.dag.task.total_running', value=1, count=1)
 
 
-def test_dag_task_ongoing_duration(aggregator, task_instance):
-    instance = common.FULL_CONFIG['instances'][0]
+@pytest.mark.parametrize(
+    "collect_ongoing_duration, expected_metric_count, should_call_method",
+    [
+        (True, 1, True),  # `collect_task_duration=True`: metric is collected, method is called
+        (False, 0, False),  # `collect_task_duration=False`: metric is NOT collected, method is NOT called
```

Comment:
```suggestion
        pytest.param(True, 1, [mock.call()], id="collect"),
        pytest.param(False, 0, [], id="don't collect"),
```

# Motivation

1. Using pytest IDs makes it easier to spot the test if it fails. It basically takes your inline comment and puts it into the pytest log.
2. Having mock calls as a list simplifies the test implementation. Separate comment about that.

Reply:
This makes sense - thanks for the suggestion.